### PR TITLE
Fix oauth application management rake tasks.

### PR DESCRIPTION
**Pull request by lakshmivyas**
- Add more comments to explain the purpose of the file and tasks
- Refactor to make sure applications can be created from blank slate
- Fix application names to match the branding guidelines

\cc @jpslav

### CODE REVIEW

File: lib/tasks/add_ost_apps.rake [hunk 1]
```diff
@@ -17,23 +27,30 @@ namespace :accounts do
...
           auth = Authentication.find_or_create_by_uid(user.identity.id.to_s) do |new_auth|
             new_auth.provider = 'identity'
             new_auth.user_id = user.id
             new_auth.save!
           end
           # For each app above, create and assign defaults if it doesn't exist
+          cb_path = "accounts/auth/openstax/callback"
           app_data.each do |app|
-            url = "https://#{app[:prefix]}#{suffix}.openstax.org/accounts/auth/openstax/callback"
+            # If the suffix parameter starts with `http` assume that
```
Comment (Dantemss) [inline lib/tasks/add_ost_apps.rake]: Why is this necessary?

Comment (lakshmivyas) [inline lib/tasks/add_ost_apps.rake]: On vagrant installations, we are serving from a non-standard port (port forwarding on host).  In those cases, we need to provide the full-url with the non-standard port.

**Comment by lakshmivyas** [comment]
Closing and reopening the PR to trigger travis. (failed because of rubygems server issues).
